syft/pkg/cataloger/cpe_specificity.go

```diff
@@ -1,31 +1,46 @@
 package cataloger
 
-import "github.com/facebookincubator/nvdtools/wfn"
+import (
+	"sort"
+
+	"github.com/facebookincubator/nvdtools/wfn"
+)
+
+var _ sort.Interface = (*ByCPESpecificity)(nil)
 
 type ByCPESpecificity []wfn.Attributes
 
-// Implementing sort.Interface
 func (c ByCPESpecificity) Len() int      { return len(c) }
 func (c ByCPESpecificity) Swap(i, j int) { c[i], c[j] = c[j], c[i] }
 func (c ByCPESpecificity) Less(i, j int) bool {
-	return countSpecifiedFields(c[i]) > countSpecifiedFields(c[j])
+	iScore := countSpecifiedFields(c[i])
+	jScore := countSpecifiedFields(c[j])
+	if iScore == jScore {
+		return countFieldLength(c[i]) > countFieldLength(c[j])
+	}
+	return iScore > jScore
+}
+
+func countFieldLength(cpe wfn.Attributes) int {
+	return len(cpe.Part + cpe.Vendor + cpe.Product + cpe.Version + cpe.TargetSW)
 }
 
 func countSpecifiedFields(cpe wfn.Attributes) int {
-	checksForSpecifiedField := []func(cpe wfn.Attributes) bool{
-		func(cpe wfn.Attributes) bool { return cpe.Part != "" },
-		func(cpe wfn.Attributes) bool { return cpe.Product != "" },
-		func(cpe wfn.Attributes) bool { return cpe.Vendor != "" },
-		func(cpe wfn.Attributes) bool { return cpe.Version != "" },
-		func(cpe wfn.Attributes) bool { return cpe.TargetSW != "" },
+	checksForSpecifiedField := []func(cpe wfn.Attributes) (bool, int){
+		func(cpe wfn.Attributes) (bool, int) { return cpe.Part != "", 2 },
+		func(cpe wfn.Attributes) (bool, int) { return cpe.Vendor != "", 3 },
+		func(cpe wfn.Attributes) (bool, int) { return cpe.Product != "", 4 },
+		func(cpe wfn.Attributes) (bool, int) { return cpe.Version != "", 1 },
+		func(cpe wfn.Attributes) (bool, int) { return cpe.TargetSW != "", 1 },
 	}
 
-	count := 0
+	var weightedCount int
```

Comment:
I'd suggest being explicit about the initial value here, even if there's no difference in behavior. I think it helps the reader understand that we're counting, and that we may end up returning `0` here.

Reply:
I can make the update, but I don't think I follow why the assigning a 0 when a zero-value will do helps with readability.

Reply:
I appreciate this deference greatly 🙏 The change is helpful to me, and you've given me something to think about better articulating in the future. 🤔